src/Core/Context/LegacyController.php

```diff
@@ -0,0 +1,281 @@
+
+use Media;
+use Module;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Traversable;
+
+class LegacyController
+{
+    // Dependency container
+    protected ContainerInterface $container;
+
+    public ?string $className;
+
+    /**
+     * List of CSS files.
+     *
+     * @var string[]
+     */
+    public array $css_files = [];
+
+    /**
+     * List of JavaScript files.
+     *
+     * @var string[]
+     */
+    public array $js_files = [];
+
+    // List of PHP errors.
+    public array $php_errors = [];
+
+    // If AJAX parameter is detected in request, set this flag to true.
+    public bool $ajax = false;
+
+    // Controller type. Possible values: 'front', 'modulefront', 'admin', 'moduleadmin'.
+    public string $controller_type;
+
+    // Controller name.
+    public string $php_self;
+
+    public ?Module $module;
+
+    /**
+     * Errors displayed after post processing
+     *
+     * @var array<string|int, string|bool>
+     */
+    public array $errors = [];
+
+    public string $path;
+
+    public string $currentIndex;
+
+    public string $content;
+
+    public array $warnings = [];
+
+    public array $informations = [];
+
+    public array $confirmations = [];
+
+    public string|false $shopShareDatas = false;
+
+    public array $_languages = [];
+
+    public int $default_form_language;
+
+    public int $allow_employee_form_lang;
+
+    public string $layout = 'layout.tpl';
+
+    public bool $bootstrap = false;
+
+    public string $template = 'content.tpl';
+
+    public string $table = 'configuration';
+
+    public ?string $list_id;
+
+    public array $tabAccess;
+
+    public int $id = -1;
+
+    public bool $required_database = false;
+
+    // Security token
+    public ?string $token;
+
+    // "shop" or "group_shop"
+    public ?string $shopLinkType;
+
+    public array $tpl_option_vars = [];
+
+    public array $tpl_view_vars = [];
+
+    public array $tpl_required_fields_vars = [];
+
+    public ?string $base_tpl_view = null;
+
+    public ?string $base_tpl_form = null;
+
+    // If you want more fieldsets in the form
+    public bool $multiple_fieldsets = false;
+
+    public array $fields_value = [];
+
+    public ?int $max_image_size = null;
+
+    // Automatically join language table if true
+    public bool $lang = false;
+
+    // Cache for translations
+    public array $cache_lang = [];
+
+    // Required_fields to display in the Required Fields form
+    public array $required_fields = [];
+
+    public string $override_folder;
+
+    // Name and directory where class image are located
+    public array $fieldImageSettings = [];
+
+    // Image type
+    public string $imageType = 'jpg';
+
+    // Current controller name without suffix
+    public string $controller_name;
+
+    public int $multishop_context = -1;
+
+    public bool $multishop_context_group = true;
+
+    // Bootstrap variable
+    public bool $show_page_header_toolbar = false;
```

Comment:
You kept a lot of public variables that will likely never be used in a Symfony context, especially with the Symfony layout that, hopefully, will become the only layout in migrated page in v9

So I wonder if we should strip this legacy bridge from useless variables, by useless I mean if:
- we don't populate the value so that the legacy code still is aware of a contextual value that has been initialized by the Symfony controllers
- we never use the value that could have been modified by hooks during the process and check the updated values at some point of the process (like js files, CSS files for example)

Then these fields would only be like dead code, never read, never written Or maybe written but without any impact or backward compatibility then they become mostly dead code that only prevent some PHP deprecation warning of accessing a non-declared field no?

Maybe this warning is reason enough to keep the field available, I'm questioning. But then at least we should probably document which variables are now useless and which ones still have meanings and some backyard compatibility policy behind them.